### PR TITLE
Completed Random Forest Models

### PR DESCRIPTION
**Pull request by g-christodoulou**
I completed the random forest models. This data set is very large so I used ranger for better speed. The data set is very large so I also needed to balance the data for this method to work better. I trying using SMOTE (Synthetic Minority Oversampling Technique) for balancing but it was running for an extremely long time and never actually executed so I used class weights. The accuracy metrics in my second model were much better but the AUC difference from train to test was too high, suggesting overfitting. I wanted to try AUC PR (area under the precision and recall curve) but I couldn't figure it out quick enough. I will continue to see in the AUC PR tells a different story.

### CODE REVIEW

**Review by adambushman — CHANGES_REQUESTED** [review]
Hi Georgia, nice work! The "test" AUCs are pretty decent; right around mid-70s. I see what you mean about gab in metrics from training/testing. Definitely some oppotunity to tune the models.

Why don't you try splitting up the training data into many pieces and doing cross validation against a handful of hyperparameters (trees, min.depth, etc.) using a tuning grid (in caret, the `trControl()` function).

Here's a good [video](https://youtu.be/rTphHY9Chgk?si=0dznwnuhHDLEZW6a&t=437) on the subject

Then, we'd want to compare precision, recall, AUC, etc.,, to find the best version.

**Review by g-christodoulou — COMMENTED** [review]
updated the script with the smote models

**Review by adambushman — APPROVED** [review]
Looks good Georgia. Merging!
